src/main/java/org/dbos/apiary/postgres/PostgresConnection.java

```diff
@@ -158,6 +164,14 @@ public FunctionOutput callFunction(String functionName, WorkerContext workerCont
                 }
                 if (valid) {
                     ctxt.conn.commit();
+                    if (ApiaryConfig.isolationLevel == ApiaryConfig.READ_COMMITTED) {
+                        for (String secondary : ctxt.secondaryWrittenKeys.keySet()) {
+                            Map<String, List<String>> writtenKeys = ctxt.secondaryWrittenKeys.get(secondary);
+                            if (!writtenKeys.isEmpty()) {
+                                ctxt.workerContext.getSecondaryConnection(secondary).rcCommit(writtenKeys, ctxt.txc);
+                            }
+                        }
+                    }
```

Comment:
Does this part guarantee atomicity across primary and secondary data stores? E.g., Postgres is committed while the secondary hasn't committed. So the new queries won't be able to see the latest committed version. But I assume it depends on the definition of read committed in cross-DB transactions.

Reply:
That depends on what you mean by atomicity, but the anomaly you're describing is allowed under read committed isolation.